### DISCUSSION TITLE
Integrate TOM TOM live traffic data to valhalla engine

### DISCUSSION THREAD

**Discussion by NitinMehra0019** [valhalla/valhalla#4525] [Q&A]
Hi everyone, I am doing a routing project so let me give you the problem statement so I have installed instance of valhalla routing server on local machine and till now I am just calling valhalla route api to get the route between 2 points alright, so now I want to integrate live traffic source preferably from TOM TOM now can any one please give me a detailed guide how to do it and you know influence the static route and change route.

**Comment by nilsnolde** [comment]
> now can any one please give me a detailed guide how to do it 

there's many threads in this repo asking about the same thing. even a few explaining more. also I wrote https://gis-ops.com/traffic-in-valhalla. it's far from trivial, a detailed guide to this is asking us to spend many hours writing docs, which is frankly lowest possible priority. especially given that it's about integrating _commercial_ datasets.

but that's just my 2c, maybe someone else is more agreeable:) given how many times this came up though, I wouldn't hold my breath.
